zanata-adapter-glossary/src/main/java/org/zanata/adapter/glossary/GlossaryCSVReader.java

```diff
@@ -150,43 +145,17 @@ private void validateCSVEntries(@Nonnull List<String[]> entries) {
         return localeColMap;
     }
 
-    private Map<Integer, String> setupDescMap(List<String[]> entries) {
-        Map<Integer, String> descMap = new HashMap<Integer, String>();
+    /**
+     * Read last 2 columns in CSV:
+     * {source locale},{locale},{locale}...,pos,description
+     *
+     * @param entries
+     */
+    private Map<String, Integer> setupDescMap(List<String[]> entries) {
+        Map<String, Integer> descMap = new HashMap<String, Integer>();
         String[] headerRow = entries.get(0);
-
-        for (int row = 0; row < headerRow.length; row++) {
-            for (String optsHeader : customColNames) {
-                if (optsHeader.equals(headerRow[row])) {
-                    descMap.put(row, headerRow[row]);
-                }
-            }
-        }
-        /*
-         * Sort out description map according to the value (header name)
-         */
-        ValueComparator bvc = new ValueComparator(descMap);
-        TreeMap<Integer, String> sorted_map = new TreeMap<Integer, String>(bvc);
-        sorted_map.putAll(descMap);
-
-        return sorted_map;
-    }
-
-   static class ValueComparator implements Comparator<Integer>, Serializable {
-
-      Map<Integer, String> base;
-      public ValueComparator(Map<Integer, String> base) {
-          this.base = base;
-      }
-
-      public int compare(Integer a, Integer b) {
-         String strA = base.get(a);
-         String strB = base.get(b);
-
-         if (strA == null || strB == null) {
-            return (strA == null) ? -1 : 1;
-         }
-
-        return strA.compareTo(strB);
-      }
+        descMap.put(POS, headerRow.length - 2);
```

Comment:
Do we have any checks to validate the length of headerRow? In case someone tries to run with invalid file